app/models/pull.rb

```diff
@@ -188,14 +188,19 @@ class Pull < ActiveRecord::Base
     state :cannot_be_merged
   end
 
+  # If you change this condition, please review the text `text_pull_broken`.
+  def broken?
+    ! repository
```

Comment:
In addition to this check, you should check if the physical repository exists (maybe by running Git commands on it).